### DISCUSSION TITLE
Is there any problem with the object containing the upper canvas

### DISCUSSION THREAD

**Discussion by rkawkxkdwlq** [fabricjs/fabric.js#9128] [Q&A]
When you create a clipPath or add an element to canvas, if you look inside the object, it contains canvas.
The canvas contains the clipPath or element and continues to contain recursively.
Is there no problem with this?
I'm wondering if there's a performance problem.

![image](https://github.com/fabricjs/fabric.js/assets/24672968/0aa6a011-2ff1-499e-ac09-fc21e6780e39)

**Comment by ShaMan123** [comment]
No problem
JS supports circular trees
This is why we have toObject / toJSON
